### PR TITLE
Bundleservice test

### LINKED ISSUE #2052: Write tests for the bundleservices branch

### PR DESCRIPTION
**Pull request by hatched**
Fixes #2052

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6894/

File: jujugui/static/gui/src/app/jujulib/test-bundleservice.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      assert.equal(typeof args[7], 'function');
+      // Call the handler
+      args[7]({
+        currentTarget: {
+          response: 'not json'
+        }
+      });
+      assert.deepEqual(
+        callback.args[0],
+        ['Unable to parse response data for bundle yaml', undefined]);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/jujulib/test-bundleservice.js]: I think we normally return `null` instead of `undefined` for these methods.

Comment (hatched) [inline jujugui/static/gui/src/app/jujulib/test-bundleservice.js]: It's returning the value of the changes parse which in the case of the error is undefined.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**